packages/langium/src/lsp/document-symbol-provider.ts

```diff
@@ -48,7 +48,7 @@ export class DefaultDocumentSymbolProvider implements DocumentSymbolProvider {
             const name = this.nameProvider.getName(astNode);
             return [{
                 kind: this.nodeKindProvider.getSymbolKind(astNode),
-                name: name ?? nameNode.text,
+                name: name === undefined || name.length === 0 ? nameNode.text : name,
```

Comment:
IMO we can just replace this with `||` since that catches `undefined` and `length === 0`, due to how falsyness operates on strings.
```suggestion
                name: name || nameNode.text,
```